satellite/satellitedb/accounting.go

```diff
@@ -39,14 +39,34 @@ func (db *accountingDB) ProjectBandwidthTotal(ctx context.Context, bucketID []by
 
 // ProjectStorageTotals returns the current inline and remote storage usage for a projectID
 func (db *accountingDB) ProjectStorageTotals(ctx context.Context, projectID uuid.UUID) (int64, int64, error) {
-	rollup, err := db.db.First_BucketStorageTally_By_ProjectId_OrderBy_Desc_IntervalStart(
+
+	// Find the interval_start for the most recent tally for this project
+	latestRollup, err := db.db.First_BucketStorageTally_By_ProjectId_OrderBy_Desc_IntervalStart(
 		ctx,
 		dbx.BucketStorageTally_ProjectId(projectID[:]),
 	)
-	if err != nil || rollup == nil {
+	if err != nil || latestRollup == nil {
 		return 0, 0, err
 	}
-	return int64(rollup.Inline), int64(rollup.Remote), err
+
+	// Sum all the inline and remote values for the project that all share the same interval_start.
+	// All records for a project that have the same interval start are part of the same tally run.
+	// This should represent the most recent calculation of a project's total at rest storage.
+	var inlineSum *int64
```

Comment:
Do these need to be pointers?

Reply:
see: https://github.com/storj/storj/pull/1586#discussion_r269587590

Reply:
while pointers definitely work here, they unfortunately must get heap allocated for this to work. it's sliiiightly better to use the sql.NullInt64 type here. so

```
type inlineSum sql.NullInt64
....Scan(&inlineSum)
if err != nil || !inlineSum.Valid { ...
```

Reply:
thanks for the tip jt, this `sql.NullInt64` is cool